### PR TITLE
feat: adjust crop color based on weather

### PR DESCRIPTION
**Pull request by taibohess**

### CODE REVIEW

**Comment by DanielSwarup** [comment]
Failing test

**Comment by wifimug** [comment]
Line 13 `set_model_stage()` should either be removed or added as a function within growth_manager

File: src/controllers/crop_controller.py [hunk 1]
```diff
@@ -105,6 +113,18 @@ def procedural_generation_seed_setter(self):
...
+            growth_manager = GrowthManager(self.config, crop, self.days_per_stage)
+            planting_date = self.config["planting_date"]
+            lat = self.config["latitude"]
+            lon = self.config["longitude"]
+            barley_type = self.config["barley_type"]
+            api_key = os.environ["WEATHER_API"]
+            weather_controller = WeatherController(api_key)
+            weather_data = weather_controller.get_merged_weather_data(barley_type, planting_date, lat, lon)
+            health_status = growth_manager.evaluate_plant_health(weather_data)
+            print(health_status)
```
Comment (DanielSwarup) [inline src/controllers/crop_controller.py:126]: ideally, remove this print statement

File: src/controllers/crop_controller.py [hunk 2]
```diff
@@ -105,6 +113,18 @@ def procedural_generation_seed_setter(self):
...
+            crop.set_color(self.crop_health[health_status])
```
Comment (DanielSwarup) [inline src/controllers/crop_controller.py:127]: this is twice

**Review by DanielSwarup — APPROVED** [review]
LGTM!

**Review by Hyridex — APPROVED** [review]
LGTM

**Pull request merged**